### PR TITLE
Node: `util.promisify`: widen callback error parameter to `unknown`

### PR DESCRIPTION
**Pull request by OliverJAsh**
An error can be of any type, not just `Error | null`—similar to how you can `throw` any type. For example:

```ts
const fetch = (callback: (error: string | null, value: string) => void) => callback(null, 'yay');
```

Under the current types, when this function (or any function that has callback error parameter whose type is wider than `Error | null`) is ran through `promisify`, it does not return the correct type:

```ts
// Type '(arg1: (error: string | null, value: string) => void) => Promise<string>' is not assignable to type '() => Promise<string>'.ts(2322)
const result: () => Promise<string> = util.promisify(fetch);
```

This PR fixes that that.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/node/util.d.ts [hunk 1]
```diff
@@ -83,24 +83,27 @@ declare module "util" {
         fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6) => Promise<TResult>
     ): (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6, callback: (err: NodeJS.ErrnoException | null, result: TResult) => void) => void;
 
+    // The value type here is a "poor man's `unknown`". When these types support TypeScript
+    // 3.0+, we can replace this with `unknown`.
+    type PoorMansUnknown = {} | null | undefined;
```
Comment (OliverJAsh) [inline types/node/util.d.ts]: I thought about avoiding the need for this by making this change to the 3.1+ types, but I think we ought to provide this bug fix to all users.

**Comment by OliverJAsh** [comment]
I don't understand why the build is failing:

```
Error: Error: Command failed: npm install --ignore-scripts --no-shrinkwrap --no-package-lock --no-bin-links --no-save
npm ERR! code ENOENT
npm ERR! syscall lchown
npm ERR! path /home/vsts/.npm/_locks/staging-737dbd926de1c3da.lock
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lchown '/home/vsts/.npm/_locks/staging-737dbd926de1c3da.lock'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```

https://dev.azure.com/definitelytyped/DefinitelyTyped/_build/results?buildId=18816

**Comment by SimonSchick** [comment]
Didn't a similar change cause issues before? cc @Flarna
I recall some issues raised after the last introduction of `PoorMansUnknown` in another place.
Also going to lean out of the window here and ask what library it is that is calling back with non-errors.

**Comment by OliverJAsh** [comment]
> ask what library it is that is calling back with non-errors

I don't have one. I was writing types for a module, and when it came to defining the error type, I decided to use `unknown` because it's not safe to make assumptions. Then, when I came to promisify this function, I ran into this.

**Comment by Flarna** [comment]
The `PoorMansUnknown` was added in querystring via #33997 - but not as named type.
It's also used in `UnhandledRejectionListener` (again not as named type).
I think there were complains after changing `UnhandledRejectionListener` from using `any` to `PoorMansUnknown` as it was breaking code from some people - `any` fits all wheras `PoorMansUnknown` fits nothing.

`Promise.catch` accepts a `(reason: any) => TResult` according to lib.es5.d.ts so somehow `any` would match better than `PoorMansUnknown`.
I think for the proposed changes in this PR it's actually don't care what's used as the error type in the callbacks is not used anywhere therefore a type allowing anything should be fine - why not `any`?

I think this PR has also no potential to break anything. Or do I overlook something?

**Comment by OliverJAsh** [comment]
I tend to prefer `unknown` over `any` because the latter disables type checking and in some situations it can leak into your code, e.g. via inference:

```ts
type Callback = (error: any, result: string) => void;
const myCallback: Callback = (error, result) => {
  error.property.that.does.not.exist // runtime error!
}
```

**Comment by Flarna** [comment]
I agree that `unknown` is in general a better choice than `any` but I don't see the scenario here where an `any` could leak.

**Comment by OliverJAsh** [comment]
Maybe not in this case, as the callback will never be inlined as in my example. 🤔

If we were to use `unknown`, how might it break people's code in an undesirable way?

**Comment by SimonSchick** [comment]
Also want to point out there are linting rules for `no-any` which are very good at catching these things, it's not _perfect_ but I'm willing to say it catches 95% of all unwanted `any`s.

**Comment by Flarna** [comment]
If we would have the real `unknown` here than I would clearly favor it.
We have seen confusions regarding `PoorMansUnknown` in the past therefore I slightly tend towards `any` - but I will for sure not block this.

File: types/node/util.d.ts (hunk 1 above)
Comment (Flarna) [reply types/node/util.d.ts]: Maybe move the `PoorMansUnknown` to NodeJS namespace in global.d.ts and use it also at other locations.

**Review by Flarna — APPROVED** [review]

**Pull request merged**